xl-v1-utils.go

```diff
@@ -23,6 +23,21 @@ import (
 	"time"
 )
 
+// Returns nil even if one of the slice elements is nil.
+// Else returns the error in the first slice element.
```

Comment:
We could explain what 'reduceErrs' do this is not explaining the non-obvious of why it is done as such. 

```
reduceErrs()
``` 

reduces the slice of error elements into a singleton error. Also we could modify this to return only the error where all elements have returned the same error.  Current function is non-deterministic as you would see while writing a unit test.

A unit test is needed here.